packages/uma-auth-client/src/utils/defineWebComponent.tsx

```diff
@@ -0,0 +1,37 @@
+import { ThemeProvider } from "@emotion/react";
+import { themes } from "@lightsparkdev/ui/styles/themes";
+import React from "react";
+import ReactDOM from "react-dom/client";
+import { GlobalStyles } from "src/GlobalStyles";
+
+export default function defineWebComponent(
+  tagName: string,
+  Component: React.FC,
+) {
+  customElements.define(
```

Comment:
Would [lit element](https://lit.dev/) be helpful at all here? It seems like we're still using react inside the web component here. Maybe that's fine, but was wondering if the inverse relationship might also make sense instead (wrap the web component in a react component if you're using react). Otherwise, non-react users would need to pull in react to use our library.

Reply:
once the web component build is made, the person using it doesn't need to install react themselves if that was a concern. The 2nd PR has a couple different build paths for this reason. If we are concerned with the size of react in the built js file though then maybe using lit makes sense here but definitely seems like it would be more work to build the whole UI from scratch using that (rather than using our existing ui lib)

Reply:
OK cool, let's not over-optimize too early then on this.